src/lib/formatter/SarifFormatter.ts

```diff
@@ -27,12 +28,12 @@ abstract class SarifFormatter {
 	 * The initial json object that contains basic information such as the engine name.
 	 * This object will be added to as the violations are processed.
 	 */
-	private readonly jsonTemplate: unknown;
+	private readonly jsonTemplate: Run;
 	protected readonly engine: string;
 
-	constructor(engine: string, toolJson: unknown) {
+	protected constructor(engine: string, runJson: Run) {
```

Comment:
Abstract classes can have `protected` constructors. Parameter renamed because `Tool` is a different type exported by `@types/sarif`, and I didn't want there to be any confusion.